crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -18,85 +18,88 @@ pub(super) fn place_comment<'a>(
+    handle_parenthesized_comment(comment, locator)
+        .or_else(|comment| handle_end_of_line_comment_around_body(comment, locator))
+        .or_else(|comment| handle_own_line_comment_around_body(comment, locator))
+        .or_else(|comment| match comment.enclosing_node() {
+            AnyNodeRef::Parameters(arguments) => {
+                handle_parameters_separator_comment(comment, arguments, locator)
+                    .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator))
+            }
+            AnyNodeRef::Arguments(_) | AnyNodeRef::TypeParams(_) => {
+                handle_bracketed_end_of_line_comment(comment, locator)
+            }
+            AnyNodeRef::Comprehension(comprehension) => {
+                handle_comprehension_comment(comment, comprehension, locator)
+            }
+            AnyNodeRef::ExprAttribute(attribute) => handle_attribute_comment(comment, attribute),
+            AnyNodeRef::ExprBinOp(binary_expression) => {
+                handle_trailing_binary_expression_left_or_operator_comment(
+                    comment,
+                    binary_expression,
+                    locator,
+                )
+            }
+            AnyNodeRef::Keyword(_) => handle_dict_unpacking_comment(comment, locator),
+            AnyNodeRef::ExprDict(_) => handle_dict_unpacking_comment(comment, locator)
+                .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator)),
+            AnyNodeRef::ExprIfExp(expr_if) => handle_expr_if_comment(comment, expr_if, locator),
+            AnyNodeRef::ExprSlice(expr_slice) => {
                 handle_slice_comments(comment, expr_slice, locator)
-            } else {
-                CommentPlacement::Default(comment)
             }
-        }
-        AnyNodeRef::ModModule(_) => {
-            handle_module_level_own_line_comment_before_class_or_function_comment(comment, locator)
-        }
-        AnyNodeRef::WithItem(_) => handle_with_item_comment(comment, locator),
-        AnyNodeRef::StmtFunctionDef(function_def) => {
-            handle_leading_function_with_decorators_comment(comment)
-                .or_else(|comment| handle_leading_returns_comment(comment, function_def))
-        }
-        AnyNodeRef::StmtClassDef(class_def) => {
-            handle_leading_class_with_decorators_comment(comment, class_def)
-        }
-        AnyNodeRef::StmtImportFrom(import_from) => handle_import_from_comment(comment, import_from),
-        AnyNodeRef::StmtWith(with_) => handle_with_comment(comment, with_),
-        AnyNodeRef::ExprConstant(_) => {
-            if let Some(AnyNodeRef::ExprFString(fstring)) = comment.enclosing_parent() {
-                CommentPlacement::dangling(fstring, comment)
-            } else {
-                CommentPlacement::Default(comment)
+            AnyNodeRef::ExprStarred(starred) => {
+                handle_trailing_expression_starred_star_end_of_line_comment(comment, starred)
```

Comment:
It looks like the function is still necessary, otherwise it gets marked as a leading comment of the starred thing rather than a leading comment of the `Expr::Starred` itself. But the first two conditions aren't necessary, I don't think.